### ISSUE TITLE
Connector to MQTT host fails with bad username/password, other clients connect normally

### ISSUE THREAD

**Issue by phidauex** [emqx/emqx#13214]
### What happened?

When creating an MQTT connector to a SenseCap broker, EMQX fails to connect with error `400 TEST_FAILED: Bad username or password` while other clients (Mosquitto and MQTT Connector) connect normally.

The connector does work with test servers such as `broker.emqx.io`, so the base functionality is working, and other EMQX features are working normally (I've been running this instance for over a year).

### What did you expect to happen?

MQTT connector connects to the broker.

### How can we reproduce it (as minimally and precisely as possible)?

Unsure what is different here - either an issue with the broker itself, or an issue with the length of the client ID, username, and password, all of which are quite long (19 char username, 64 char password, 20 char min client ID). I have not yet tested a separate broker with a long password to see if it responds in a similar way. If it were helpful I can provide read-only API key over private message.

### Anything else we need to know?

Mosquitto connection command that is working normally:

```
mosquitto_sub \
    -h sensecap-openstream.seeed.cc \
    -t '/device_sensor_data/<15-char-numerical-ID>/+/+/+/+' \
    -u 'org-<15-char-numerical-ID>' \
    -P '<64 character hexidecimal API key>' \
    -I 'org-<15-char-numerical-ID>-<random>' \
    -v
```

EMQX Connector settings:
![image](https://github.com/emqx/emqx/assets/49047/85e83879-c609-40a9-b901-894e8bdaaece)

### EMQX version

<details>

```console
sysdescr  : EMQX
version   : 5.7.0
datetime  : 2024-06-10T03:29:08.905402357+00:00
uptime    : 22 hours, 55 minutes, 32 seconds
```

</details>

### OS version

<details>

```console
$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
$ uname -a
Linux emqx 6.8.4-2-pve #1 SMP PREEMPT_DYNAMIC PMX 6.8.4-2 (2024-04-10T17:36Z) x86_64 x86_64 x86_64 GNU/Linux
```

</details>

### Log files

<details>

```console
2024-06-10T03:31:38.483960+00:00 [error] msg: start_ecpool_error, reason: {malformed_username_or_password,undefined}, pool_name: <<"connector:mqtt:SenseCap">>
2024-06-10T03:31:38.484005+00:00 [warning] msg: start_resource_failed, id: <<"connector:mqtt:SenseCap">>, reason: {malformed_username_or_password,undefined}
2024-06-10T03:31:53.553148+00:00 [warning] msg: ingress_client_connect_failed, name: <<"connector:mqtt:SenseCap">>, reason: {malformed_username_or_password,undefined}
2024-06-10T03:31:53.553294+00:00 [error] crasher: initial call: ecpool_worker:init/1, pid: <0.55262.0>, registered_name: [], exit: {{malformed_username_or_password,undefined},[{gen_server,init_it,6,[{file,"gen_server.erl"},{line,961}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,241}]}]}, ancestors: [<0.55261.0>,<0.55259.0>,ecpool_sup,<0.2951.0>], message_queue_len: 1, messages: [{'EXIT',<0.55263.0>,{shutdown,malformed_username_or_password}}], links: [<0.55261.0>], dictionary: [], trap_exit: true, status: running, heap_size: 1598, stack_size: 28, reductions: 2846; neighbours:
2024-06-10T03:31:53.553421+00:00 [error] Supervisor: {<0.55261.0>,ecpool_worker_sup}. Context: start_error. Reason: {malformed_username_or_password,undefined}. Offender: id={worker,1},pid=undefined.
2024-06-10T03:31:53.553482+00:00 [error] Supervisor: {<0.55259.0>,ecpool_pool_sup}. Context: start_error. Reason: {shutdown,{failed_to_start_child,{worker,1},{malformed_username_or_password,undefined}}}. Offender: id=worker_sup,pid=undefined.
2024-06-10T03:31:53.553684+00:00 [error] msg: start_ecpool_error, reason: {malformed_username_or_password,undefined}, pool_name: <<"connector:mqtt:SenseCap">>
2024-06-10T03:31:53.553726+00:00 [warning] msg: start_resource_failed, id: <<"connector:mqtt:SenseCap">>, reason: {malformed_username_or_password,undefined}
2024-06-10T03:32:08.624868+00:00 [warning] msg: ingress_client_connect_failed, name: <<"connector:mqtt:SenseCap">>, reason: {malformed_username_or_password,undefined}
2024-06-10T03:32:08.625007+00:00 [error] crasher: initial call: ecpool_worker:init/1, pid: <0.55274.0>, registered_name: [], exit: {{malformed_username_or_password,undefined},[{gen_server,init_it,6,[{file,"gen_server.erl"},{line,961}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,241}]}]}, ancestors: [<0.55273.0>,<0.55271.0>,ecpool_sup,<0.2951.0>], message_queue_len: 1, messages: [{'EXIT',<0.55275.0>,{shutdown,malformed_username_or_password}}], links: [<0.55273.0>], dictionary: [], trap_exit: true, status: running, heap_size: 1598, stack_size: 28, reductions: 2846; neighbours:
2024-06-10T03:32:08.625153+00:00 [error] Supervisor: {<0.55273.0>,ecpool_worker_sup}. Context: start_error. Reason: {malformed_username_or_password,undefined}. Offender: id={worker,1},pid=undefined.
2024-06-10T03:32:08.625215+00:00 [error] Supervisor: {<0.55271.0>,ecpool_pool_sup}. Context: start_error. Reason: {shutdown,{failed_to_start_child,{worker,1},{malformed_username_or_password,undefined}}}. Offender: id=worker_sup,pid=undefined.
2024-06-10T03:32:08.625426+00:00 [error] msg: start_ecpool_error, reason: {malformed_username_or_password,undefined}, pool_name: <<"connector:mqtt:SenseCap">>
2024-06-10T03:32:08.625472+00:00 [warning] msg: start_resource_failed, id: <<"connector:mqtt:SenseCap">>, reason: {malformed_username_or_password,undefined}
```

</details>

**Comment by zmstone** [comment]
According to the doc https://sensecap.gitbook.io/doc/data-openstream-api/api-reference
It might expect the client ID to follow its scheme.
EMQX tries to keep the client ID less than 33 bytes, when the length exceeds 32 bytes, the concatenated client ID is hashed to a 32 bytes space.

**Comment by zmstone** [comment]
Will try to fix it in 5.7.1
